src/widgets/index/index.ts

```diff
@@ -402,11 +439,25 @@ const index = (props: IndexProps): Index => {
       // widgets. When the subscription happens before the `init` step, the (static)
       // configuration of the widget is pushed in the URL. That's what we want to avoid.
       // https://github.com/algolia/instantsearch.js/pull/994/commits/4a672ae3fd78809e213de0368549ef12e9dc9454
-      helper.on('change', ({ state }) => {
-        localUiState = getLocalWidgetsState(localWidgets, {
-          searchParameters: state,
-          helper: helper!,
-        });
+      helper.on('change', event => {
+        const { state, isPageReset } = event;
+
+        // @ts-ignore _isFromAddWidget comes from privateHelperSetState and
+        // thus isn't typed on the helper event
+        const _isFromAddWidget = event._isFromAddWidget;
+
+        if (isPageReset) {
+          localUiState.page = undefined;
+        }
+
+        localUiState = getLocalWidgetsState(
+          localWidgets,
+          {
+            searchParameters: state,
+            helper: helper!,
+          },
+          _isFromAddWidget ? localUiState : {}
```

Comment:
```suggestion
          // @MAJOR in a next version we can always use localUiState, instead of usually empty object), but it requires every single widget to set an empty value (or remove the state key's value) if they are not refined.
          _isFromAddWidget ? localUiState : {}
```